src/plugins/sources/azureservicebus.js

```diff
@@ -0,0 +1,91 @@
+var request = require('request'),
+  util = require('util'),
+  assert = require('assert'),
+  crypto = require('crypto'),
+  parseString = require('xml2js').parseString;
+
+var AzureServiceBusSource = function() {
+
+  var source = {};
+
+  // Function which creates the Service Bus SAS token.
+  var getToken = function(uri, sasKeyName, sasKey) {
+    var endocedResourceUri = encodeURIComponent(uri);
+    var t0 = new Date(1970, 1, 1, 0, 0, 0, 0);
+    var t1 = new Date();
+    var expireInSeconds = +(31 * 24 * 3600) + 3600 +
+      (((t1.getTime() - t0.getTime()) / 1000) | 0);
+    var plainSignature = (endocedResourceUri + '\n' + expireInSeconds).toString('utf8');
+    var hash = crypto.createHmac('sha256', sasKey).update(plainSignature).digest('base64');
+    var token = 'SharedAccessSignature sr=' + endocedResourceUri + '&sig=' +
+      encodeURIComponent(hash) + '&se=' + expireInSeconds + '&skn=' +
+      sasKeyName;
+    return token;
+  };
+
+  function validate(datasource) {
```

Comment:
Ah, you call this for each datasourrce as before. Got it. Do we need the generic validator then?